airflow/contrib/hooks/aws_glue_job_hook.py

```diff
@@ -0,0 +1,203 @@
+        self.retry_limit = retry_limit or 0
+        self.num_of_dpus = num_of_dpus or 10
+        self.region_name = region_name
+
+        if s3_bucket is None:
+            raise AirflowException(
+                'Failed to create aws glue job, error: Specify Parameter `s3_bucket`'
+            )
+
+        self.s3_bucket = s3_bucket
+        self.role_name = iam_role_name
+        self.S3_PROTOCOL = "s3://"
+        self.S3_ARTIFACTS_PREFIX = 'artifacts/glue-scripts/'
+        self.S3_GLUE_LOGS = 'logs/glue-logs/'
+        super(AwsGlueJobHook, self).__init__(*args, **kwargs)
+
+    def get_conn(self):
+        conn = self.get_client_type('glue', self.region_name)
+        return conn
+
+    def list_jobs(self):
+        conn = self.get_conn()
+        return conn.get_jobs()
+
+    def get_iam_execution_role(self):
+        """
+        :return: iam role for job execution
+        """
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.role_name)
+            self.log.info("Iam Role Name: {}".format(self.role_name))
+            return glue_execution_role
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to create aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def initialize_job(self, script_arguments=None):
+        """
+        :return:
+        """
+        glue_client = self.get_conn()
+
+        try:
+            job_response = self.get_or_create_glue_job()
+            job_name = job_response['Name']
+            job_run = glue_client.start_job_run(
+                JobName=job_name,
+                Arguments=script_arguments
+            )
+            return self._job_completion(job_name, job_run['JobRunId'])
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to run aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def _job_completion(self, job_name=None, run_id=None):
+        """
+        :param job_name:
+        :param run_id:
+        :return:
+        """
+        glue_client = self.get_conn()
+        job_status = glue_client.get_job_run(
+            JobName=job_name,
+            RunId=run_id,
+            PredecessorsIncluded=True
+        )
+        job_run_state = job_status['JobRun']['JobRunState']
+        failed = job_run_state == 'FAILED'
+        stopped = job_run_state == 'STOPPED'
+        completed = job_run_state == 'SUCCEEDED'
+
+        while True:
+            if failed or stopped or completed:
+                self.log.info("Exiting Job {} Run State: {}"
+                              .format(run_id, job_run_state))
+                return job_run_state
+            else:
+                print(job_status, failed)
+                time.sleep(1000)
```

Comment:
Time sleep is seconds not microseconds

Reply:
Do you mean 1000 seconds is to much?

Reply:
Maybe :smile:

Reply:
Good. Any recommendations in mind?

Reply:
I would use 1 or 5 seconds. The aim is just to prevent the busy loop.

Reply:
@bolkedebruin do you have something better in mind? Don't really know much about Sensors or if they can be applied